lntest/itest/lnd_amp_test.go

```diff
@@ -38,6 +38,15 @@ func testSendPaymentAMPInvoiceCase(net *lntest.NetworkHarness, t *harnessTest,
 	ctx := newMppTestContext(t, net)
 	defer ctx.shutdownNodes()
 
+	// Subscribe to bob's invoices. Do this early in the test to make sure
+	// that the subscription has actually been completed when we add an
+	// invoice. Otherwise the notification will be missed.
+	req := &lnrpc.InvoiceSubscription{}
```

Comment:
I belive we could fix this below in `testSendToRouteAMP` too.

Reply:
I think that one has sufficient delay because it inserts the jit-inserts the invoice when the htlc arrives. It is kind of unsatisfactory that the subscribe call doesn't give a signal when it is actually subscribed though.